New package: lhapdf5

wdconinc opened the pull request:
LHAPDF up to version 5 is a general purpose Fortran 77/90 interpolator, used for evaluating PDFs from discretised data files.

This package is similar to HepMC2 and HepMC3 in that it is reasonable to expect both to be installed simultaneously in environments (see e.g. https://github.com/spack/spack/pull/16892 for a discussion which applies here as well).

Link: https://lhapdf.hepforge.org/lhapdf5/

(Part of an effort to contribute spack packages from the Electron-Ion Collider repository to the broader NP and HEP community.)

wdconinc committed:
[lhapdf5] New package

adamjstewart commented on var/spack/repos/builtin/packages/lhapdf5/package.py:
```diff
@@ -0,0 +1,36 @@
+from spack import *
+
+
+class Lhapdf5(Package):
```
```suggestion
class Lhapdf5(AutotoolsPackage):
```

adamjstewart commented on var/spack/repos/builtin/packages/lhapdf5/package.py:
```diff
@@ -0,0 +1,36 @@
+    def setup_build_environment(self, env):
+        env.append_flags('FFLAGS', '-std=legacy')
+
+    def install(self, spec, prefix):
+        configure("--prefix={0}".format(prefix))
+        make()
+        make('install')
```
This can be removed once you inherit from AutotoolsPackage

wdconinc committed:
[lhpadf5] Inherit from AutotoolsPackage

wdconinc committed:
[lhapdf5] Variant python2, without which python extension is disabled

During configure lhapdf5 searches for python. On one system
I tested on (ubuntu 19.10) it finds a system installed python3
and fails to create the python extension.

Variant named to make explicit that this is only a python2 extension.

wdconinc commented:
Sorry for the edit in post-review. I managed to reproduce an issue I had with this lhapdf5 version three weeks back (https://github.com/eic/eic-spack/issues/15), but that I hadn't been able to reproduce. This last commit fixes that issue.

adamjstewart commented on var/spack/repos/builtin/packages/lhapdf5/package.py:
```diff
@@ -0,0 +1,46 @@
+    version('5.8.0', sha256='8381ea5f785dde95772a2b6d5890f1cb72012e223e6861823fd81b09eedaa7a3')
+    version('5.7.1', sha256='40529629351598317fbf7b5905661e51b23778019d50451eee78d7b1118e2559')
+
+    variant('python2', default=False,
```
I would call this variant `python` to be in line with other packages in Spack.

adamjstewart commented on var/spack/repos/builtin/packages/lhapdf5/package.py:
```diff
@@ -0,0 +1,46 @@
+        if self.spec.satisfies('+python2'):
+            env.append_flags(
+                'PYTHON',
+                join_path(self.spec['python'].prefix.bin, 'python'))
```
You should use `self.spec['python'].command.path`, the executable is not always called `python`

adamjstewart commented on var/spack/repos/builtin/packages/lhapdf5/package.py:
```diff
@@ -0,0 +1,46 @@
+
+    def configure_args(self):
+        args = []
+        if self.spec.satisfies('-python2'):
```
I would explicitly enable this in an if clause (`if '+python' in self.spec`) and disable it in an else clause

wdconinc commented:
This was merged before the review and additional changes. Do you want me to file a new PR? I wanted to make sure the commit stayed with the revision requests.

adamjstewart commented:
You can open a follow-up PR, that's fine.
